src/dune_rules/expander.ml

```diff
@@ -115,19 +115,26 @@ let expand_env t pform s : Value.t list option =
       Some [ String (Option.value ~default (Env.get t.env var)) ]
 
 let expand_version scope pform s =
+  let value_from_version = function
+    | None -> [ Value.String "" ]
+    | Some s -> [ String s ]
+  in
   match
     Package.Name.Map.find
       (Dune_project.packages (Scope.project scope))
       (Package.Name.of_string s)
   with
   | None ->
-    User_error.raise
-      ~loc:(String_with_vars.Var.loc pform)
-      [ Pp.textf "Package %S doesn't exist in the current project." s ]
-  | Some p -> (
-    match p.version with
-    | None -> [ Value.String "" ]
-    | Some s -> [ String s ] )
+    begin match Lib.DB.find (Scope.libs scope) (Lib_name.of_string s) with
```

Comment:
As discussed earlier, I think it's odd to allow library names such as `foo.bar` as input. If we are returning the version of the package, we should just allow package names. So it should be `Lib_name.of_string (Package.Name.to_string (Package.Name.of_string ..)))`.

Reply:
I directly forbid the syntax in https://github.com/ocaml/dune/pull/4103/commits/71267b26d91cdd095bd13ab28677d26878c76fd2. What do you think?